### PR TITLE
Feat/email username to unique string

### PR DESCRIPTION
**Pull request by yarkhinephyo**
#### Summary
Instead of using email as identifier for the users in Cognito, we use unique usernames instead. This makes it easier to utilize user identifiers in the frontend as email is considered more sensitive.

#### Changes
- Refactoring was done for the frontend for logging in and signing up
  - No feature change except now you use username to log in
- Removal of displayName in the backend as it used to store the usernames
- Updated Postman and Swagger

### CODE REVIEW

File: ionic_user_interface/src/views/Confirm.vue [hunk 1]
```diff
@@ -105,7 +105,7 @@ export default defineComponent({
 
     onMounted(() => {
       msgBoxColor.value = 'medium';
-      msgBox.value?.showMsg(`A confirmation code has been sent to ${getUserEmail().value}`);
+      msgBox.value?.showMsg(`A confirmation code has been sent to the email`);
```
Comment (qwoprocks) [inline ionic_user_interface/src/views/Confirm.vue:108]: Is there a way to show the email? For user-friendliness

File: ionic_user_interface/src/views/Login.vue [hunk 2]
```diff
@@ -93,27 +93,26 @@ export default defineComponent({
...
     const errorMsg: Ref<typeof MessageBox | null> = ref(null);
 
     onIonViewDidLeave(() => {
       errorMsg.value?.close();
     });
 
-    const isValidEmail = (email: string): boolean => {
-      return emailPattern.test(email.toLowerCase());
+    const isValidUsername = (username: string): boolean => {
+      return username.length >= 5 && usernamePattern.test(username);
```
Comment (qwoprocks) [inline ionic_user_interface/src/views/Login.vue:115]: tbh we should set a length upper limit as well

File: ionic_user_interface/src/providers.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import jwt_decode from 'jwt-decode';
 import { toastController } from '@ionic/vue';
 
 const isLoggedIn = ref(false);
-const userEmail = ref('');
+const username = ref('');
```
Comment (qwoprocks) [inline ionic_user_interface/src/providers.ts:8]: No need for this anymore, I was using userEmail to display their email in Confirm.vue. Best if we can get the userEmail from the server when "UserNotConfirmed" is returned

File: lambda_backend/route_microservice/src/getRoutesByGym.ts [hunk 4]
```diff
@@ -16,6 +16,7 @@ const getRoutesByGym: Handler = async (event: GetRoutesByGymEvent) => {
     TableName: process.env['ROUTE_TABLE_NAME'],
     IndexName: 'gymLocationIndex',
     ConsistentRead: false,
+    ScanIndexForward: false,
```
Comment (qwoprocks) [inline lambda_backend/route_microservice/src/getRoutesByGym.ts:19]: What does this do?

**Review by qwoprocks — CHANGES_REQUESTED** [review]

File: lambda_backend/route_microservice/src/getRoutesByGym.ts (hunk 4 above)
Comment (yarkhinephyo) [reply lambda_backend/route_microservice/src/getRoutesByGym.ts:19]: Return routes by reversed sorted date (Latest routes first)

File: ionic_user_interface/src/providers.ts (hunk 3 above)
Comment (yarkhinephyo) [reply ionic_user_interface/src/providers.ts:8]: `username` is needed now as a param when confirming. I will try to return email

Comment (yarkhinephyo) [reply ionic_user_interface/src/providers.ts:8]: Okie done

**Pull request merged**